x509/x509.go

```diff
@@ -40,6 +41,13 @@ type pkixPublicKey struct {
 	BitString asn1.BitString
 }
 
+// ParsedDomainName is a structure holding a parsed domain name (CommonName or DNS SAN) and a parsing error.
+type ParsedDomainName struct {
+	DomainString string
+	ParsedDomain *publicsuffix.DomainName
```

Comment:
Does this also need to be a pointer?

Reply:
Not necessarily. The  `publicsuffix.DomainFromListWithOptions` returns a pointer, so I kept is as a pointer. I could run some benchmarks to see if it makes sense to dereference it here if you wish.